core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3322,25 +3362,40 @@ private SyncResult syncInodeMetadata(RpcContext rpcContext, LockedInodePath inod
         }
       }
       if (syncPlan.toLoadMetadata()) {
-        loadMetadata = true;
+        AlluxioURI mountUri = new AlluxioURI(mMountTable.getMountPoint(inodePath.getUri()));
+        // We only sync up to the mount point or the originalPath, whichever is lower in the tree
+        if (PathUtils.hasPrefix(originalPath.getUri().getPath(), mountUri.getPath())) {
+          pathsToLoad.add(originalPath.getUri().getPath());
+        } else {
+          pathsToLoad.add(mountUri.getPath());
+        }
       }
       if (syncPlan.toSyncChildren() && inode instanceof InodeDirectory
           && syncDescendantType != DescendantType.NONE) {
         UfsStatus[] listStatus = ufs.listStatus(ufsUri.toString());
-        if (listStatus != null) {
+
+        // if the alluxio path contains a mount point, we need to keep going to find it
+        if (listStatus != null || containsMountPoint) {
           InodeDirectory inodeDir = (InodeDirectory) inode;
           // maps children name to inode
           Map<String, Inode<?>> inodeChildren = new HashMap<>();
           for (Inode<?> child : inodeDir.getChildren()) {
             inodeChildren.put(child.getName(), child);
           }
-
-          for (UfsStatus ufsChildStatus : listStatus) {
-            if (!inodeChildren.containsKey(ufsChildStatus.getName()) && !PathUtils
-                .isTemporaryFileName(ufsChildStatus.getName())) {
-              // Ufs child exists, but Alluxio child does not. Must load metadata.
-              loadMetadata = true;
-              break;
+          if (listStatus != null) {
+            for (UfsStatus ufsChildStatus : listStatus) {
+              if (!inodeChildren.containsKey(ufsChildStatus.getName()) && !PathUtils
+                  .isTemporaryFileName(ufsChildStatus.getName())) {
+                // Ufs child exists, but Alluxio child does not. Must load metadata.
+                AlluxioURI mountUri = new AlluxioURI(mMountTable.getMountPoint(inodePath.getUri()));
+                // We only sync up to the mount point or the originalPath
+                if (PathUtils.hasPrefix(originalPath.getUri().getPath(), mountUri.getPath())) {
```

Comment:
can we just return the mount points as the paths, and then the user of this info can check to see if the mount point should be loaded, or the original inode path?

Reply:
Is the reason for doing this so that we don't need to pass down `originalPath`?

Reply:
yeah, I would like to avoid having to pass down `originalPath`.